pymc_marketing/clv/distributions.py

```diff
@@ -293,3 +294,192 @@ def logp(value, lam, p, T, T0):
             pt.all(T0 < T),
             msg="lam > 0, 0 <= p <= 1, T0 < T",
         )
+
+
+class ParetoNBDRV(RandomVariable):
+    name = "pareto_nbd"
+    ndim_supp = 1
+    ndims_params = [0, 0, 0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("ParetoNBD", "\\operatorname{ParetoNBD}")
+
+    def make_node(self, rng, size, dtype, r, alpha, s, beta, T):
+
+        r = pt.as_tensor_variable(r)
+        alpha = pt.as_tensor_variable(alpha)
+        s = pt.as_tensor_variable(s)
+        beta = pt.as_tensor_variable(beta)
+        T = pt.as_tensor_variable(T)
+
+        return super().make_node(rng, size, dtype, r, alpha, s, beta, T)
+
+    def __call__(self, r, alpha, s, beta, T, size=None, **kwargs):
+        return super().__call__(r, alpha, s, beta, T, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(cls, rng, r, alpha, s, beta, T, size) -> np.array:
+
+        size = pm.distributions.shape_utils.to_tuple(size)
+
+        r = np.asarray(r)
+        alpha = np.asarray(alpha)
+        s = np.asarray(s)
+        beta = np.asarray(beta)
+        T = np.asarray(T)
+
+        if size == ():
+            size = np.broadcast_shapes(
+                r.shape, alpha.shape, s.shape, beta.shape, T.shape
+            )
+
+        r = np.broadcast_to(r, size)
+        alpha = np.broadcast_to(alpha, size)
+        s = np.broadcast_to(s, size)
+        beta = np.broadcast_to(beta, size)
+        T = np.broadcast_to(T, size)
+
+        output = np.zeros(shape=size + (2,))
+
+        lam = rng.gamma(shape=r, scale=1 / (alpha + 10e-8), size=size)
+        mu = rng.gamma(shape=s, scale=1 / (beta + 10e-8), size=size)
+
+        def sim_data(lam, mu, T):
+            t = 0
+            n = 0
+
+            dropout_time = rng.exponential(scale=1 / (mu + 10e-8))
+            wait = rng.exponential(scale=1 / (lam + 10e-8))
+
+            while t + wait < min(dropout_time, T):
+                t += wait
+                n += 1
+                wait = rng.exponential(scale=1 / lam)
+
+            return np.array(
+                [
+                    n,
+                    t,
+                ],
+            )
```

Comment:
I think the only reason I switched this is because the `lifetimes` convention has frequency first. However, that pertains more to the utility functions and is mainly personal preference here, so I switched it back to `np.array([t, n])`.

Reply:
I did not know about the `lifetimes` convention, that makes sense. We can go with `np.array([t, n])`, as long as the other distribution classes have it in the same order